sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuOverrides.scala

```diff
@@ -3199,6 +3199,14 @@ object GpuOverrides extends Logging {
         override def convertToGpu(child: Expression): GpuExpression =
           GpuSize(child, a.legacySizeOfNull)
       }),
+    expr[Reverse](
```

Comment:
Reverse only supports arrays and string as input.

https://github.com/apache/spark/blob/89b2ee27d258dec8fe265fa862846e800a374d8e/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/collectionOperations.scala#L1225

We should not say the NULL type is supported. We should Say that Spark supports `TypeSig.ARRAY.nested(TypeSig.all)`

When we do that we don't need the incompat tag, because the TypeSig will document that types we don't support and the types that we do support. It will also be 100% compatible for strings.

Reply:
Thank you! Updated this.